README
NAME
    Carp::Always::Dump - (DEPRECATED) Like Carp::Always, but dumps the
    content of function arguments

VERSION
    This document describes version 0.07 of Carp::Always::Dump (from Perl
    distribution Carp-Always-Dump), released on 2015-01-06.

SYNOPSIS
     % perl -MCarp::Always::Dump script.pl

DESCRIPTION
    NOTICE: This module is deprecated. The same functionality is in
    Devel::Confess:

     % perl -d:Confess=dump ...

    This module will be removed from CPAN.

VARIABLES
  $Color => BOOL (default: from COLOR environment, or 1)
    If set to true, will use terminal colors to help visually distinguish
    parameters from one another.

  $DumpObj => BOOL (default: 0)
    If set to 1, will dump objects (blessed references) instead of showing
    them as 'Foo::Bar=HASH(0x19c8ff8)'.

  $MaxArgLen => INT (default: 0)
    Like $MaxArgLen in Carp, to limit the number of characters of dump to
    show.

  $Terse => BOOL (default: 1)
    If set to false, will use Data::Dump instead of the terser
    Data::Dump::OneLine to produce the dumps.

IMPORTS
    For each variable mentioned in "VARIABLES", you can also set it via
    import argument:

     use Carp::Always::Dump Color=>0, DumpObj=>1;

    Via command-line:

     % perl -MCarp::Always::Dump=Color,1 ...

ENVIRONMENT
  COLOR => BOOL
    Used to set the default of $Color.

  INTERACTIVE => BOOL
SEE ALSO
    Carp::Always (and its variants such as: Carp::Always::Color,
    Carp::Always::SHS)

    Devel::SimpleTrace, a simpler stack trace module, without showing
    function arguments.

    Monkey::Patch::Action

HOMEPAGE
    Please visit the project's homepage at
    <https://metacpan.org/release/Carp-Always-Dump>.

SOURCE
    Source repository is at
    <https://github.com/perlancar/perl-Carp-Always-Dump>.

BUGS
    Please report any bugs or feature requests on the bugtracker website
    <https://rt.cpan.org/Public/Dist/Display.html?Name=Carp-Always-Dump>

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR
    perlancar <perlancar@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2015 by perlancar@cpan.org.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Revision history for Carp-Always-Dump

0.07    2015-01-06 (PERLANCAR)

	- No functional changes.

	- Mark deprecation, removing from CPAN.


0.06    2013-11-12 (SHARYANTO)

	- No functional changes. Rebuild to fix t/00-compile.t generated by
	  older DZP::Test::Compile (2.019 < x < 2.033).


0.05    2013-10-10 (SHARYANTO)

        - Turn off color when not run interactively.


0.04    2013-08-07 (SHARYANTO)

        - Allow setting option variables via import().


0.03    2013-07-16 (SHARYANTO)

        - Add unimport() to restore Carp::Always' original behavior.


0.02    2013-07-16 (SHARYANTO)

        - Add $Terse to pick Data::Dump over Data::Dump::OneLine.


0.01    2013-07-16 (SHARYANTO)

        - First release.


Build.PL

use strict;
use warnings;

use Module::Build 0.3601;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.3601"
  },
  "configure_requires" => {
    "Module::Build" => "0.3601"
  },
  "dist_abstract" => "Like Carp::Always, but dumps the content of function arguments",
  "dist_author" => [
    "Steven Haryanto <stevenharyanto\@gmail.com>"
  ],
  "dist_name" => "Carp-Always-Dump",
  "dist_version" => "0.04",
  "license" => "perl",
  "module_name" => "Carp::Always::Dump",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp::Always" => 0,
    "Data::Dump" => 0,
    "Data::Dump::OneLine" => 0,
    "Monkey::Patch::Action" => 0,
    "perl" => "5.010001"
  },
  "script_files" => [],
  "test_requires" => {}
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  my $tr = delete $module_build_args{test_requires};
  my $br = $module_build_args{build_requires};
  for my $mod ( keys %$tr ) {
    if ( exists $br->{$mod} ) {
      $br->{$mod} = $tr->{$mod} if $tr->{$mod} > $br->{$mod};
    }
    else {
      $br->{$mod} = $tr->{$mod};
    }
  }
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;


lib/Carp/Always/Dump.pm
package Carp::Always::Dump;

use 5.010001;
use strict;
use warnings;

use Data::Dump qw(dump);
use Data::Dump::OneLine qw(dump1);
use Monkey::Patch::Action qw(patch_package);
use Scalar::Util qw(blessed);

our $VERSION = '0.07'; # VERSION

our $Color     = $ENV{COLOR} // $ENV{INTERACTIVE} // (-t STDOUT);
our $DumpObj   = 0;
our $MaxArgLen = 0;
our $Terse     = 1;

our $h;

require Carp;
require Carp::Always;

sub import {
    my ($self, %args) = @_;

    for my $k (keys %args) {
        my $v = $args{$k};
        if ($k =~ /^\$?Color$/) {
            $Color = $v;
        } elsif ($k =~ /^\$?DumpObj$/) {
            $DumpObj = $v;
        } elsif ($k =~ /^\$MaxArgLen$/) {
            $MaxArgLen = $v;
        } elsif ($k =~ /^\$?Terse$/) {
            $Terse = $v;
        } else {
            die "Unknown import argument $k, please use one of: ".
                "Color/DumpObj/MaxArgLen/Terse";
        }
    }

    $h = patch_package(
        "Carp", "format_arg", "replace",
        sub {
            my $arg = shift;
            my $res;

            if (blessed($arg) && !$DumpObj) {
                $res = "'$arg'";
            } else {
                if ($Terse) {
                    $res = dump1($arg);
                } else {
                    $res = dump($arg);
                }
                $res = substr($res, 0, $MaxArgLen) . "..."
                    if $MaxArgLen > 0 && $MaxArgLen < length($res);
            }

            state $colnum = 0;
            if ($Color) {
                my $col;
                if ($colnum++ % 2) {
                    $col = 33;
                } else {
                    $col = 36;
                }
                $res = "\e[$col;3m$res\e[0m";
            }

            return $res;
        }
    );
}

sub unimport {
    undef $h;
}

1;
# ABSTRACT: (DEPRECATED) Like Carp::Always, but dumps the content of function arguments

__END__

=pod

=encoding UTF-8

=head1 NAME

Carp::Always::Dump - (DEPRECATED) Like Carp::Always, but dumps the content of function arguments

=head1 VERSION

This document describes version 0.07 of Carp::Always::Dump (from Perl distribution Carp-Always-Dump), released on 2015-01-06.

=head1 SYNOPSIS

 % perl -MCarp::Always::Dump script.pl

=head1 DESCRIPTION

B<NOTICE:> This module is deprecated. The same functionality is in
L<Devel::Confess>:

 % perl -d:Confess=dump ...

This module will be removed from CPAN.

=head1 VARIABLES

=head2 $Color => BOOL (default: from COLOR environment, or 1)

If set to true, will use terminal colors to help visually distinguish parameters
from one another.

=head2 $DumpObj => BOOL (default: 0)

If set to 1, will dump objects (blessed references) instead of showing
them as 'Foo::Bar=HASH(0x19c8ff8)'.

=head2 $MaxArgLen => INT (default: 0)

Like C<$MaxArgLen> in L<Carp>, to limit the number of characters of dump to
show.

=head2 $Terse => BOOL (default: 1)

If set to false, will use L<Data::Dump> instead of the terser
L<Data::Dump::OneLine> to produce the dumps.

=head1 IMPORTS

For each variable mentioned in L</"VARIABLES">, you can also set it via import
argument:

 use Carp::Always::Dump Color=>0, DumpObj=>1;

Via command-line:

 % perl -MCarp::Always::Dump=Color,1 ...

=head1 ENVIRONMENT

=head2 COLOR => BOOL

Used to set the default of C<$Color>.

=head2 INTERACTIVE => BOOL

=head1 SEE ALSO

L<Carp::Always> (and its variants such as: L<Carp::Always::Color>,
L<Carp::Always::SHS>)

L<Devel::SimpleTrace>, a simpler stack trace module, without showing function
arguments.

L<Monkey::Patch::Action>

=head1 HOMEPAGE

Please visit the project's homepage at L<https://metacpan.org/release/Carp-Always-Dump>.

=head1 SOURCE

Source repository is at L<https://github.com/perlancar/perl-Carp-Always-Dump>.

=head1 BUGS

Please report any bugs or feature requests on the bugtracker website L<https://rt.cpan.org/Public/Dist/Display.html?Name=Carp-Always-Dump>

When submitting a bug or request, please include a test-file or a
patch to an existing test-file that illustrates the bug or desired
feature.

=head1 AUTHOR

perlancar <perlancar@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by perlancar@cpan.org.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.025.
use strict;
use warnings;

use 5.010001;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "(DEPRECATED) Like Carp::Always, but dumps the content of function arguments",
  "AUTHOR" => "perlancar <perlancar\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Perl::osnames" => "0.09"
  },
  "DISTNAME" => "Carp-Always-Dump",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010001",
  "NAME" => "Carp::Always::Dump",
  "PREREQ_PM" => {
    "Carp::Always" => 0,
    "Data::Dump" => 0,
    "Data::Dump::OneLine" => 0,
    "Monkey::Patch::Action" => 0
  },
  "TEST_REQUIRES" => {
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0
  },
  "VERSION" => "0.07",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp::Always" => 0,
  "Data::Dump" => 0,
  "Data::Dump::OneLine" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Monkey::Patch::Action" => 0,
  "Perl::osnames" => "0.09",
  "Test::More" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



